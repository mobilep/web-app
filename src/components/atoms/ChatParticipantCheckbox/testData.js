export default {
	user: {
		_id: '5d70bc6d93b6e912b83a65d5',
		firstName: 'Taras',
		lastName: 'Kvas',
		avatarColor: 'fd524f',
		avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
		avatar_md: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/640/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
		avatar_lg: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/1024/2e3db607-9ec1-45f4-a7e5-4bed03912dd5.jpg',
		name: 'Taras Kvas',
	},
};
