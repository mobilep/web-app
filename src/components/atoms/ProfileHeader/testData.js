export const props = {
	name: 'Andy Murray',
	company: 'Techmagic',
	initials: 'JD',
	avatarColor: '#87d4ee',
	avatar: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
	editText: 'Edit',
};
