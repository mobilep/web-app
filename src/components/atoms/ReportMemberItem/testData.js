export const userPropWithStar = {
	_id: '5d4bfe2095a0842e643c2453',
	value: 4.7,
	maxValue: 5,
	withStar: true,
	firstName: 'Alexa',
	lastName: 'Five',
	avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/533567e7-c1f2-4751-8e7b-9dd55c71305e.jpg',
	avatarColor: '6c6ced',
	name: 'Alexa Five',
	initials: 'AF',
};

export const userProp = {
	_id: '5d4bfe2095a0842e643c2453',
	value: 10.1,
	firstName: 'Alexa',
	lastName: 'Five',
	avatar_sm: 'https://s3-eu-west-1.amazonaws.com/mobilepractice-photo/public/100/533567e7-c1f2-4751-8e7b-9dd55c71305e.jpg',
	avatarColor: '6c6ced',
	name: 'Alexa Five',
	initials: 'AF',
};
