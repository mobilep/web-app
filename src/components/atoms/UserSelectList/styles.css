.UserSelectList {
	width: 100%;
	padding: 4px 0;
	background-color: var(--white);
	box-shadow: 0 2px 4px 0 var(--transparentGrey1);
}

.UserSelectList-item {
	display: flex;
	width: 100%;
	align-items: center;
	height: 32px;
	padding: 0 12px;
	margin: 0;
	background-color: var(--white);
	border: none;
	will-change: background-color;
	transition: background-color .2s ease-in-out;
	cursor: pointer;
}

.UserSelectList-item-active {
	background-color: var(--grey2);
}

.UserLabel.UserSelectList-userLabel {
	border-radius: 0;
	width: 100%;
	background-color: transparent;
	padding: 4px 0;
}

.UserSelectList-notFound {
	margin: 4px 0;
	padding: 4px;
}

.UserSelectList-spinnerWrapper {
	display: flex;
	flex-direction: column;
	justify-items: center;
}

.UserSelectList-spinnerWrapper svg {
	height: 32px;
	width: 32px;
}

.UserSelectList-notFound:hover {
	cursor: pointer;
	background-color: var(--grey2);
}
