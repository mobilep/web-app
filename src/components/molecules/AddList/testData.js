export const users = [
	{
		id: 1,
		fullName: 'John Smith',
		initials: 'JS',
	}, {
		id: 2,
		fullName: 'Philippe Starck',
		initials: 'PS',
	}, {
		id: 3,
		fullName: 'Suzy Salmon',
		initials: 'SS',
	},
];

export const allUsers = [
	{
		id: 4,
		fullName: 'John Rebele',
		initials: 'JR',
	}, {
		id: 5,
		fullName: 'John Klokdan',
		initials: 'JK',
	}, {
		id: 6,
		fullName: 'John Makeba',
		initials: 'JM',
	},
];
