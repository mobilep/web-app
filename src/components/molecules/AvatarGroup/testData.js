export const avatars = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];

export const avatarsLess = [
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
	{
		firstName: 'Clayton',
		lastName: 'Gross',
		img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQBlUvtjoDyRYSmS6hoSVke1HcE-3ZeLGHLR7RnTIBp3jAYv39B',
		avatarColor: '#87d4ee',
	},
];
