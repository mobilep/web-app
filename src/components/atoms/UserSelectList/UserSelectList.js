import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { UserLabel, Spinner } from '..';

import './styles.css';

const getClassName = (className) => classNames('UserSelectList', className);

const UserSelectList = ({
	items = [],
	onSelect,
	className,
	selectedItem,
	onMouseOver,
	styles = {},
	notFound,
	userSearchFetching,
	notFoundMessage,
	selectInput,
}) => {
	const handleClick = (id) => () => onSelect(id);
	const handleMouseOver = (index) => () => onMouseOver(index);

	const getItems = () => {
		if (userSearchFetching) {
			return (
				<div className='UserSelectList-spinnerWrapper'>
					<Spinner color='rgb(243, 66, 60)' />
				</div>
			);
		}
		if (notFound) {
			return (
				<div className='UserSelectList-notFound'>
					{notFoundMessage}
				</div>
			);
		} else {
			selectInput();
			return items.map(({ id, fullName, ...rest }, index) => {
				const key = `selectItem${index}`;
				const className = selectedItem === index
					? 'UserSelectList-item UserSelectList-item-active'
					: 'UserSelectList-item';
				return (
					<button
						className={className}
						key={key}
						onMouseOver={handleMouseOver(index)}
						onMouseLeave={handleMouseOver(null)}
						onClick={handleClick(id)}
					>
						<UserLabel className='UserSelectList-userLabel'
							fullName={fullName.split(' ').reverse().join(' ')}
							{...rest} />
					</button>
				);
			});
		}
	};
	return (
		<div className={getClassName(className)} style={styles}>
			{getItems()}
		</div>
	);
};

export default UserSelectList;

UserSelectList.defaultProps = {
	onMouseOver: () => {},
	onSelect: () => {},
};

UserSelectList.propTypes = {
	className: PropTypes.string,
	items: PropTypes.array,
	notFound: PropTypes.bool,
	notFoundMessage: PropTypes.string,
	onMouseOver: PropTypes.func,
	onSelect: PropTypes.func,
	selectInput: PropTypes.func,
	selectedItem: PropTypes.number,
	styles: PropTypes.object,
	userSearchFetching: PropTypes.bool,
};
