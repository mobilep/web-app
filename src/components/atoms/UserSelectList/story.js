import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import UserSelectList from './UserSelectList';
import { items } from './testData';

storiesOf('1. Atoms / UserSelectList', module)
	.add('UserSelectList', () => (
		<UserSelectList items={items} onSelect={action('onSelect')} />
	));
