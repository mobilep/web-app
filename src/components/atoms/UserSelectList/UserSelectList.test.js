import React from 'react';
import { shallow } from 'enzyme';

import UserSelectList from './UserSelectList';
import { items } from './testData';

describe('Atom - UserSelectList', () => {

	it('Should render', () => {
		const wrapper = shallow(<UserSelectList
			items={items}
			selectInput={() => {}}
		/>);
		expect(wrapper).toMatchSnapshot();
	});

});
