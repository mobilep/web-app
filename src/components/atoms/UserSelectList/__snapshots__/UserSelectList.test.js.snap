// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom - UserSelectList Should render 1`] = `
<div
  className="UserSelectList"
  style={Object {}}
>
  <button
    className="UserSelectList-item"
    key="selectItem0"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <UserLabel
      className="UserSelectList-userLabel"
      fullName="Rebele John"
      initials="JR"
    />
  </button>
  <button
    className="UserSelectList-item"
    key="selectItem1"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <UserLabel
      className="UserSelectList-userLabel"
      fullName="Ðšlokdam John"
      initials="JK"
    />
  </button>
  <button
    className="UserSelectList-item"
    key="selectItem2"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <UserLabel
      className="UserSelectList-userLabel"
      fullName="Makeba John"
      initials="JM"
    />
  </button>
</div>
`;
